src/Security/GroupCsvBulkLoader.php

```diff
@@ -33,9 +33,9 @@ public function processRecord($record, $columnMap, &$results, $preview = false)
         // are imported to avoid missing "early" references to parents
         // which are imported later on in the CSV file.
         if (isset($record['ParentCode']) && $record['ParentCode']) {
-            $parentGroup = Group::get()->setUseCache(true)->find('Code', $record['ParentCode']);
-            if ($parentGroup) {
-                $group->ParentID = $parentGroup->ID;
+            $parentGroupIDs = Group::get()->setUseCache(true)->filter('Code', $record['ParentCode'])->column();
+            if (!empty($parentGroupIDs)) {
+                $group->ParentID = $parentGroupIDs[0];
```

Comment:
Instead of getting all the data about the group just to get its ID, we just get the ID directly.